x-pack/platform/plugins/shared/streams/server/lib/streams/client.ts

```diff
@@ -207,26 +215,126 @@ export class StreamsClient {
+          query,
+        })),
+      ]
+    );
 
-    await this.dependencies.assetClient.syncAssetList(stream.name, [
-      ...dashboards.map((dashboard) => ({
-        [ASSET_ID]: dashboard,
-        [ASSET_TYPE]: 'dashboard' as const,
-      })),
-      ...queryLinks,
-    ]);
+    await this.manageQueries(name, {
+      deleted: deleted.filter(isQueryLink),
+      indexed: indexed.filter(isQueryLink),
+      updated: updated.filter(isQueryLink),
+    });
 
     return {
       acknowledged: true,
       result: result.changes.created.includes(name) ? 'created' : 'updated',
     };
   }
 
+  async manageQueries(
+    name: string,
+    {
+      deleted = [],
+      indexed = [],
+      updated = [],
+    }: { deleted?: QueryLink[]; indexed?: QueryLink[]; updated?: QueryLink[] }
+  ) {
+    if (this.dependencies.config.experimental?.significantEventsEnabled !== true) {
+      return;
+    }
+
+    await Promise.all([
+      this.installQueries(name, { indexed, updated }),
+      this.uninstallQueries(name, deleted),
+    ]);
+  }
+
+  private async installQueries(
+    name: string,
+    { indexed = [], updated = [] }: { indexed: QueryLink[]; updated: QueryLink[] }
+  ) {
+    const { rulesClient } = this.dependencies;
+
+    const indexedIds = new Set(indexed.map(getRuleIdFromQueryLink));
+    const updatedIds = new Set(updated.map(getRuleIdFromQueryLink));
+
+    const toUninstall = updated.filter((query) => !indexedIds.has(getRuleIdFromQueryLink(query)));
+    const toUpdate = indexed.filter((query) => updatedIds.has(getRuleIdFromQueryLink(query)));
+    const toCreate = indexed.filter((query) => !updatedIds.has(getRuleIdFromQueryLink(query)));
+
+    await Promise.all([
+      this.uninstallQueries(name, toUninstall),
+      ...toUpdate.map((query) => {
+        const ruleId = getRuleIdFromQueryLink(query);
+        return rulesClient.update<EsqlRuleParams>({
+          id: ruleId,
+          data: {
+            name: query.query.title,
+            actions: [],
+            params: {
+              timestampField: '@timestamp',
+              query: `FROM ${name},${name}.* METADATA _id, _source | WHERE KQL(\"\"\"${query.query.kql.query}\"\"\")`,
+            },
+            tags: ['streams'],
+            schedule: {
+              interval: '1m',
+            },
+          },
+        });
+      }),
+      ...toCreate.map((query) => {
+        const ruleId = getRuleIdFromQueryLink(query);
+        return rulesClient.create<EsqlRuleParams>({
+          data: {
+            name: query.query.title,
+            consumer: 'streams',
+            alertTypeId: 'streams.rules.esql',
+            actions: [],
+            params: {
+              timestampField: '@timestamp',
+              query: `FROM ${name},${name}.* METADATA _id, _source | WHERE KQL(\"\"\"${query.query.kql.query}\"\"\")`,
+            },
+            enabled: true,
+            tags: ['streams'],
+            schedule: {
+              interval: '1m',
```

Comment:
1m or 2m? (I thought we wanted to go with 2m but not sure)

Reply:
I don't recall honestly, but also with the lowered 1k limit, I think it's better to do more smaller runs. But happy to change to 2m if you think other-way